subprojects/signing/src/integTest/groovy/org/gradle/plugins/signing/SigningPublicationsIntegrationSpec.groovy

```diff
@@ -304,8 +378,10 @@ class SigningPublicationsIntegrationSpec extends SigningIntegrationSpec {
                 sign publishing.publications.mavenJava
             }
 
-            signMavenJavaPublication.signatures.removeAll { signature ->
-                signature.toSign.name.endsWith('.jar')
+            signMavenJavaPublication.signatures.all { signature ->
+                if (signature.toSign.name.endsWith('.jar')) {
+                    signMavenJavaPublication.signatures.remove signature
+                }    
```

Comment:
Curious... why was this change necessary?

Reply:
Because this was assuming eager creation of signatures, which is no longer the case now that `from(component)` is lazy.